front/pages/api/w/[wId]/spaces/[spaceId]/data_sources/index.ts

```diff
@@ -507,6 +521,53 @@ const handleDataSourceWithProvider = async ({
 
   await dataSource.setConnectorId(connectorsRes.value.id);
 
+  // For Slack apps, register the signing secret in the webhook router
+  if (provider === "slack" && connectionId) {
+    const oauthAPI = new OAuthAPI(config.getOAuthAPIConfig(), logger);
+    const accessTokenRes = await oauthAPI.getAccessToken({ connectionId });
+    if (accessTokenRes.isOk()) {
+      const connection = accessTokenRes.value.connection;
+      const signingSecret = await getSlackSigningSecret(oauthAPI, connectionId);
```

Comment:
Not a fan of defining a whole new endpoint for this and be able to retrieve a secret from just a connectionId. With my [PR](https://github.com/dust-tt/dust/pull/18492/files) merged you should be able to use the related_credential_id set in the connection response and use it to query the credential by id of Oauth directly. wdyt ?

Reply:
done